projects/plugins/boost/app/modules/cache/storage/File_Storage.php

```diff
@@ -102,4 +102,57 @@ private function sanitize_path( $path ) {
 		$_cache[ $path ] = $path;
 		return $path;
 	}
+
+	/**
+	 * Delete all cached data for the given path.
+	 *
+	 * @param string $path - The path to delete.
+	 */
+	public function invalidate( $path ) {
+		error_log( "invalidate: $path" ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+		$path = $this->sanitize_path( $path );
+		$dir  = $this->root_path . $path;
+
+		if ( Boost_Cache_Utils::is_boost_cache_directory( $dir ) ) {
+			return Boost_Cache_Utils::delete_directory( $dir );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Given a request_uri and its parameters, delete the cached data for this request.
+	 *
+	 * @param string $request_uri - The URI of this request (excluding GET parameters)
+	 * @param array  $parameters  - An associative array of all the things that make this request special/different. Includes GET parameters and COOKIEs normally.
+	 */
+	public function invalidate_single_visitor( $request_uri, $parameters ) {
+		$directory = self::get_uri_directory( $request_uri );
+		$filename  = self::get_request_filename( $request_uri, $parameters );
+		$full_path = $directory . $filename;
+		error_log( 'Deleting ' . $full_path . ' from cache' ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+
+		if ( file_exists( $full_path ) ) {
+			return wp_delete_file( $full_path );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Delete the cached files for the home page, and any paged archives.
+	 */
+	public function invalidate_home_page() {
+		$directory = $this->root_path;
+		$dir       = Boost_Cache_Utils::sanitize_file_path( $directory );
+
+		if ( Boost_Cache_Utils::is_boost_cache_directory( $dir ) ) {
+			if ( is_dir( $dir . '/page' ) ) {
+				Boost_Cache_Utils::delete_directory( $dir . '/page' );
+			}
+			return Boost_Cache_Utils::delete_single_directory( $dir );
+		}
+
+		return false;
+	}
```

Comment:
If WordPress is installed in a subdir, or is otherwise not at the `/` URL of their site, then won't this completely fail to clear the cache of their homepage?

Also what is the check for `/page` for? I think that needs a comment explaining what it means.

Reply:
You're right about a subdirectory. I'm going to test that. /page/ is for paged archives. Here's [page 426](https://inphotos.org/page/426/) of my photoblog. :)

Reply:
I just fixed that.